docs/source/reference/kubernetes/kubernetes-debugging.rst

```diff
@@ -0,0 +1,280 @@
+
+Does your cluster support LoadBalancer services?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To verify that your cluster supports LoadBalancer services, we will create an example service and verify that it gets an external IP.
+
+.. code-block:: bash
+
+    $ kubectl apply -f https://raw.githubusercontent.com/skypilot-org/skypilot/master/tests/kubernetes/cpu_test_pod.yaml
+    $ kubectl apply -f https://raw.githubusercontent.com/skypilot-org/skypilot/master/tests/kubernetes/loadbalancer_test_svc.yaml
+
+    # Verify that the service gets an external IP
+    # Note: It may take some time on cloud providers to change from pending to an external IP
+    $ watch kubectl get svc skytest-loadbalancer
+
+    # Once you get an IP, try accessing the HTTP server by curling the external IP
+    $ IP=$(kubectl get svc skytest-loadbalancer -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    $ curl $IP:8080
+
+    # Once you have verified that the service is accessible, you can delete it
+    $ kubectl delete -f https://raw.githubusercontent.com/skypilot-org/skypilot/master/tests/kubernetes/cpu_test_pod.yaml
+    $ kubectl delete -f https://raw.githubusercontent.com/skypilot-org/skypilot/master/tests/kubernetes/loadbalancer_test_svc.yaml
+
+If your service does not get an external IP, check the service's status with :code:`kubectl describe svc skytest-loadbalancer`. Your cluster may not support LoadBalancer services.
+
+
+.. _kubernetes-debugging-ports-nginx:
+
+Step C0 - Verifying Nginx Ingress setup
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you are using Nginx Ingress for ports support, refer to :ref:`kubernetes-ingress` for instructions on how to install and configure Nginx Ingress.
+
+.. tip::
+
+    If you are using LoadBalancer services for ports support, you can skip this section.
+
+Does your cluster support Nginx Ingress?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To verify that your cluster supports Nginx Ingress, we will create an example ingress.
+
+
+Is SkyPilot configured to use Nginx Ingress?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Take a look at your :code:`~/.sky/config.yaml` file to verify that the :code:`ports: ingress` section is configured correctly.
+
+.. code-block:: bash
+
+    $ cat ~/.sky/config.yaml
+
+    # Output should contain:
+    #
+    # kubernetes:
+    #   ports: ingress
+
+If not, add the :code:`ports: ingress` section to your :code:`~/.sky/config.yaml` file.
+
+.. _kubernetes-debugging-ports-dryrun:
+
+Step C1 - Verifying SkyPilot can launch services
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Next, try running a simple task with a service to verify that SkyPilot can launch services on your cluster.
+
+.. code-block:: bash
+
+    $ sky launch -y -c myserver --cloud kubernetes --port 8080 -- "python -m http.server 8080"
+
+    # Obtain the endpoint of the service
+    $ sky status --endpoint 8080 myserver
```

Comment:
When I run this after my cluster is up, I got the following error:
```
sky status --endpoint 8080 myserver
RuntimeError: Port 8080 not exposed yet. If the cluster was recently started, please retry after a while. Additionally, make sure your Ingress is configured correctly.
To debug, run: kubectl describe ingress && kubectl describe ingressclass
```
I run the two commands suggested, but fail to see what should I look at in the output.

In the `kubectl describe ingress`, I do see the ingress setup:
```
kubectl describe ingress
Name:             myserver-084e-skypilot-ingress--8080
Labels:           <none>
Namespace:        default
Address:          10.166.0.4,10.166.0.5
Ingress Class:    nginx
Default backend:  <default>
Rules:
  Host        Path  Backends
  ----        ----  --------
  *           
              /skypilot/myserver-084e/8080(/|$)(.*)   myserver-084e-skypilot-service--8080:8080 (10.42.167.175:8080)
Annotations:  field.cattle.io/publicEndpoints:
                [{"addresses":["10.166.0.4","10.166.0.5"],"port":80,"protocol":"HTTP","serviceName":"default:myserver-084e-skypilot-service--8080","ingres...
              nginx.ingress.kubernetes.io/rewrite-target: /$2
              nginx.ingress.kubernetes.io/use-regex: true
Events:
  Type    Reason  Age                   From                      Message
  ----    ------  ----                  ----                      -------
  Normal  Sync    6m2s (x3 over 6m23s)  nginx-ingress-controller  Scheduled for sync
  Normal  Sync    6m2s (x3 over 6m23s)  nginx-ingress-controller  Scheduled for sync
```
However, the `sky status --endpoint 8080 myserver` still shows the same error. 

I also tried `kubectl describe services` and seems the service for the nginx ingress is there:
```
Name:              myserver-084e-skypilot-service--8080
Namespace:         default
Labels:            parent=skypilot
Annotations:       <none>
Selector:          skypilot-cluster=myserver-084e
Type:              ClusterIP
IP Family Policy:  SingleStack
IP Families:       IPv4
IP:                10.43.98.142
IPs:               10.43.98.142
Port:              <unset>  8080/TCP
TargetPort:        8080/TCP
Endpoints:         10.42.167.175:8080
Session Affinity:  None
Events:            <none>
```

```

Reply:
Just tried to install the ingress controller with the quick start's recommendation, but got an error.
```
helm upgrade --install ingress-nginx ingress-nginx \
  --repo https://kubernetes.github.io/ingress-nginx \
  --namespace ingress-nginx --create-namespace
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/azureuser/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/azureuser/.kube/config
Release "ingress-nginx" does not exist. Installing it now.
Error: Unable to continue with install: IngressClass "nginx" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "ingress-nginx": current value is "rke2-ingress-nginx"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "ingress-nginx": current value is "kube-system"
```

Reply:
Ah looks like the default RKE nginx ingress doesn't have a service configured to access the ingress controller.

The [Bare-Metal instructions](https://kubernetes.github.io/ingress-nginx/deploy/#bare-metal-clusters) on the nginx installation docs work here, tested on RKE2 cluster. I'll update the docs.

Reply:
Added a tip - let me know if that helps!